Rubberduck.CodeAnalysis/CodePathAnalysis/Nodes/Implementations/ReferenceNode.cs

```diff
@@ -4,6 +4,11 @@ namespace Rubberduck.Inspections.CodePathAnalysis.Nodes
 {
     public class ReferenceNode : NodeBase
     {
-        public ReferenceNode(IParseTree tree) : base(tree) { }
+        public ReferenceNode(IParseTree tree, AssignmentNode value) : base(tree)
+        {
+            ValueNode = value;
+        }
+
+        public AssignmentNode ValueNode { get; }
```

Comment:
Note: not used for `AssignmentNotUsedInspection` - remove?

Reply:
Given that this is the vanilla code path analysis API, I would definitely want to remove. I think the API itself needs to be made generic, which an inspection can then specialize for their own uses to keep the concerns clean, no?

Reply:
IDK, maybe I'm misunderstanding what the code path analysis API is supposed to do, but the way I see it, it's very much its job to create things like `ReferenceNode` and `AssignmentNode` - where `AssignmentNode` has its `Usages` already tracked, therefore where `ReferenceNode` already knows what assignment/value it's referring to, i.e. the client code (e.g. inspections) doesn't need to further process the trees much.
I'll still remove it because YAGNI as far as this PR is concerned, but I feel like the boundaries of this API need to be discussed/clarified.